apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+
+  // when sending an sms to several recipients in activity, we'll exit the
+  // activity after this delay after moving to the thread list.
+  LEAVE_ACTIVITY_DELAY: 3000,
+
+  draft: null,
+  recipients: null,
+  // Set to |true| when in edit mode
+  inEditMode: false,
+  isNewMessageNoticeShown: false,
+  shouldChangePanelNextEvent: false,
+  showErrorInFailedEvent: '',
+  previousSegment: 0,
+
+  timeouts: {
+    update: null,
+    subjectLengthNotice: null
+  },
+
+  multiSimActionButton: null,
+  init: function conv_init() {
+    var templateIds = [
+      'message',
+      'message-sim-information',
+      'message-status',
+      'not-downloaded',
+      'recipient',
+      'date-group',
+      'header',
+      'group-header'
+    ];
+
+    // Fields with 'messages' label
+    [
+      'container', 'to-field', 'recipients-list', 'compose-form', 'header',
+      'edit-header', 'check-uncheck-all-button', 'contact-pick-button',
+      'send-button', 'delete-button', 'call-number-button', 'options-button',
+      'new-message-notice', 'edit-mode', 'edit-form', 'header-text',
+      'max-length-notice', 'convert-notice', 'resize-notice',
+      'new-message-notice', 'subject-max-length-notice', 'sms-counter-notice',
+      'recipient-suggestions'
+    ].forEach(function(id) {
+      this[Utils.camelCase(id)] = document.getElementById('messages-' + id);
+    }, this);
+
+    this.mainWrapper = document.getElementById('main-wrapper');
+    this.threadMessages = document.getElementById('thread-messages');
+
+    window.addEventListener('resize', this.resizeHandler.bind(this));
+
+    // binding so that we can remove this listener later
+    this.onVisibilityChange = this.onVisibilityChange.bind(this);
+    document.addEventListener('visibilitychange',
+                              this.onVisibilityChange);
+
+    this.toField.addEventListener(
+      'keypress', this.toFieldKeypress.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'input', this.toFieldInput.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'focus', this.toFieldInput.bind(this), true
+    );
+
+    this.sendButton.addEventListener(
+      'click', this.onSendClick.bind(this)
+    );
+
+    this.sendButton.addEventListener(
+      'contextmenu', this.onSendClick.bind(this)
+    );
+
+    this.container.addEventListener(
+      'scroll', this.manageScroll.bind(this)
+    );
+
+    this.editHeader.addEventListener(
+      'action', this.cancelEdit.bind(this)
+    );
+
+    this.header.addEventListener(
+      'action', this.onHeaderAction.bind(this)
+    );
+
+    this.optionsButton.addEventListener(
+      'click', this.showOptions.bind(this)
+    );
+
+    this.callNumberButton.addEventListener('click', function() {
+      ActivityPicker.dial(Threads.active.participants[0]);
+    });
+
+    this.deleteButton.addEventListener(
+      'click', this.delete.bind(this)
+    );
+
+    this.headerText.addEventListener(
```

Comment:
```headerText``` is managed by both views, but the logic is quite different, so I think it can be just separated.